pkg/agent/ipassigner/ip_assigner_linux.go

```diff
@@ -33,6 +33,15 @@ import (
 	"antrea.io/antrea/pkg/agent/util/sysctl"
 )
 
+var (
+	netlinkAddrAdd = netlink.AddrAdd
+	netlinkAddrDel = netlink.AddrDel
+)
+
+type advertiseArpNdp func(a *ipAssigner, ip net.IP)
+
+var advertiseResponder advertiseArpNdp = (*ipAssigner).advertise
```

Comment:
This is less readable and harder to mock/restore than declaring them as member variables of `ipAssigner`.

```
type ipAssigner struct {
        ...

	netlink netlink.Interface
	advertiseFn func(ip net.IP)
}
```
Then they can mocked when testing without being converted to receiver-less functions, and no need to be restored after tests.

Reply:
ipAssigner structure is used at many places, so it can avoid touching existing code for UT  !

advertise func is a local function of ip_assigner_linux.go which acts on ipAssigner structure,  advertise func is called internally from AssignIP, we are testing AssignIP and need to mock  advertise before calling AssignIP from test, it can only work if we define it a type as
 func(a *ipAssigner, ip net.IP) !

similarly for external pkg apis like netlink.AddrAdd and  netlink.AddrDel are called from inside of  AssignIP and we needed to mock their behavior before calling AssignIP !

You meant here to include internal function advertise in ipAssigner structure and then mock it before calling AssignIP from test ?